arch/x86/include/x86_mmu.h

```diff
@@ -209,7 +209,7 @@ extern pentry_t z_x86_kernel_ptables[];
 static inline pentry_t *z_x86_thread_page_tables_get(struct k_thread *thread)
 {
 #if defined(CONFIG_USERSPACE) && !defined(CONFIG_X86_COMMON_PAGE_TABLE)
-	if (!IS_ENABLED(CONFIG_X86_KPTI) ||
+	if (!(IS_ENABLED(CONFIG_X86_KPTI)) ||
```

Comment:
Would it make sense to change the `IS_ENABLED` macro itself instead of all of its usages?

Reply:
Unfortunately IS_ENABLED is used also in macros that need it expands only to bare 0 or 1.

Reply:
Maybe there could be two macros? One for boolean and one for number. I am not sure this makes sense.

Reply:
If there are not many places that rely on the expansion to bare 0 or 1 then I think two macros should be better. It looks quite awkward to have to put parentheses around IS_ENABLED() in if-statements.